### PR TITLE
Fix Infinite loop on OdbcDataReader.internalGetString with UTF-8 text

### PR DESCRIPTION
**Pull request by MohamedChaawa**
Fix Infinite loop on OdbcDataReader.internalGetString with UTF-8 text  
Issue: https://github.com/dotnet/runtime/issues/86490

### CODE REVIEW

**Comment by MohamedChaawa** [comment]
Can you please review this PR? @SamMonoRT @roji   @ajcvickers

File: src/libraries/System.Data.Odbc/src/System/Data/Odbc/OdbcDataReader.cs [hunk 1]
```diff
@@ -1008,6 +1008,11 @@ private object internalGetString(int i)
                             // followed by exception. I did not add it now to avoid breaking change
                             Debug.Assert(lengthOrIndicator >= 0 || lengthOrIndicator == ODBC32.SQL_NO_TOTAL, "unexpected lengthOrIndicator value");
 
+                            if (0 == lengthOrIndicator)
+                            {
+                                break;  // done
+                            }
```
Comment (stephentoub) [inline src/libraries/System.Data.Odbc/src/System/Data/Odbc/OdbcDataReader.cs:1014]: Is there any way to add a test for this?

**Comment by MohamedChaawa** [comment]
@roji @AndriySvyryd Any news about this pullrequest review please?
This bug is blocking our projet development advancements.

**Comment by roji** [comment]
@MohamedChaawa can you please address the comment above about adding a test?

File: src/libraries/System.Data.Odbc/src/System/Data/Odbc/OdbcDataReader.cs [hunk 2]
```diff
@@ -1008,6 +1008,11 @@ private object internalGetString(int i)
                             // followed by exception. I did not add it now to avoid breaking change
                             Debug.Assert(lengthOrIndicator >= 0 || lengthOrIndicator == ODBC32.SQL_NO_TOTAL, "unexpected lengthOrIndicator value");
 
+                            if (0 == lengthOrIndicator)
```
Comment (danmoseley) [inline src/libraries/System.Data.Odbc/src/System/Data/Odbc/OdbcDataReader.cs:1011]:
```suggestion
                            if (lengthOrIndicator == 0)
```

**Pull request closed without merging**